src/main/java/net/journey/entity/projectile/EntitySentryKingGrenade.java

```diff
@@ -29,25 +30,74 @@ public boolean apply(@Nullable Entity e) {
 	public EntitySentryKingGrenade(World worldIn) {
 		super(worldIn);
 		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = 200;
+
+	}
+	
+	public EntitySentryKingGrenade(World worldIn, int grenadeTime) {
+		super(worldIn);
+		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = grenadeTime;
 
 	}
 
 	public EntitySentryKingGrenade(World worldIn, EntityLivingBase shooter) {
 		super(worldIn, shooter);
 		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = 200;
+
+	}
+	
+	public EntitySentryKingGrenade(World worldIn, EntityLivingBase shooter, int grenadeTime) {
+		super(worldIn, shooter);
+		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = grenadeTime;
 
 	}
 
 	public EntitySentryKingGrenade(World worldIn, double x, double y, double z) {
 		super(worldIn, x, y, z);
 		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = 200;
+
+	}
+	
+	public EntitySentryKingGrenade(World worldIn, double x, double y, double z, int grenadeTime) {
+		super(worldIn, x, y, z);
+		this.pickupStatus = PickupStatus.DISALLOWED;
+		this.grenadeTimer = grenadeTime;
 
 	}
 
 	public EntitySentryKingGrenade(World worldIn, EntityLivingBase e, EntityLivingBase eb, float f, float f1) {
 		super(worldIn);
 		Entity.setRenderDistanceWeight(10.0D);
 		this.shootingEntity = e;
+		this.grenadeTimer = 200;
+
+		this.posY = e.posY + e.getEyeHeight() - 0.10000000149011612D;
```

Comment:
shooting in constructor is a bad way of doing things.
In constructor you're initialize the object, without any actions